iso.bbx

```diff
@@ -114,87 +135,181 @@
+    {\mainlstring{urlfrom}}%
+  \addcolon\space\url{#1}%
 }
+
+% \mkbibacro typesets acronyms in small caps by default,
+% if we want to override this, pass the value as is (uppercase)
+% \renewcommand*{\mkbibacro}[1]{#1}
+
+% Format DOI identifier as hyperlink (possibly)
 \DeclareFieldFormat{doi}{%
-  \printtext{DOI}\addcolon\space
+  \mkbibacro{DOI}\addcolon\space
   \ifhyperref
     {\href{https://doi.org/#1}{\nolinkurl{#1}}}
     {\nolinkurl{#1}}%
 }
+
+% Format medium type in square brackets
 \DeclareFieldFormat{howpublished}{\mkbibbrackets{#1}}
+% If it is online entry type, defaults to [online]
 \DeclareFieldFormat[online]{howpublished}{%
   \iffieldundef{howpublished}
     {\mkbibbrackets{online}}
     {\mkbibbrackets{#1}}%
 }
 
+% Format type, e.g. type of thesis
+% If available, use the localisation string for the key
 \DeclareFieldFormat{type}{%
   \ifbibstring{#1}
     {\mainlstring{#1}}
     {#1}%
 }
 
+% Format supervisor of thesis (precedes the name with a localisation string)
 \DeclareFieldFormat{supervisor}{%
   \mainlstring{bysupervisor}\addspace#1}
 
-%We must handle more than one isbn and issn
+% We must handle more than one isbn and issn
 \DeclareFieldFormat{isbn}{%
-  %\renewcommand*{\do}[1]{\printtext{ISBN}\addspace#1\adddot\addspace}%
-  \printtext{ISBN}\addspace#1}%
+  %\renewcommand*{\do}[1]{\mkbibacro{ISBN}\addspace#1\adddot\space}%
+  \mkbibacro{ISBN}\addspace#1}%
 
 \DeclareFieldFormat{issn}{%
-  \renewcommand*{\do}[1]{\printtext{ISSN}\addspace#1\adddot\addspace}%
+  \renewcommand*{\do}[1]{\mkbibacro{ISSN}\addspace#1\adddot\space}%
   \docsvfield{issn}}%
 
-\DeclareFieldFormat{isan}{\printtext{ISAN}\addspace#1}
-\DeclareFieldFormat{ismn}{\printtext{ISMN}\addspace#1}
-\DeclareFieldFormat{isrn}{\printtext{ISRN}\addspace#1}
-\DeclareFieldFormat{iswc}{\printtext{ISWC}\addspace#1}
+% Format other standard identifiers, precedes the value with the name
+% of identifier
+\DeclareFieldFormat{isan}{\mkbibacro{ISAN}\addspace#1}
+\DeclareFieldFormat{ismn}{\mkbibacro{ISMN}\addspace#1}
+\DeclareFieldFormat{isrn}{\mkbibacro{ISRN}\addspace#1}
+\DeclareFieldFormat{iswc}{\mkbibacro{ISWC}\addspace#1}
 
+% Format url seen date with preceding localisation string in square brackets
 \DeclareFieldFormat{urldate}{\mkbibbrackets{\mainsstring{urlseen}\space#1}}
 
+% a)
+% Format circa date with preceding localisation string in square brackets;
+% default \datecircaprint is \bibstring{circa}\printdelim{datecircadelim},
+% e.g. [ca. 1940].
+% b)
+% Adds non-breaking space after open date range, e.g. '1492-- '.
+% c)
+% If there is no date field in the bib entry,
+% we should state there is no date by printing
+% a localisation string, e.g. [n.d.].
+\DeclareFieldFormat{date}{%
+  % Check any end year for open date range
+  \ifboolexpr{
+    test {\iffieldequalstr{endyear}{}}
+    or test {\iffieldequalstr{eventendyear}{}}
+    or test {\iffieldequalstr{origendyear}{}}
+  }
+    {\ifdatecirca
+      {\mkbibbrackets{#1\mbox{\addnbspace}}}%
```

Comment:
I'm wondering if the space after the dash in an open-ended date range is really necessary. If it is, I would probably find it nicer to code the space into `\mkdaterangefull` and friends. (Of course that is more painful, but it feels right.)

If it turns out that the test should remain here, I'm wondering if there should really be an 'or' here. It appears to me that now the `evendate` field has the potential to influence the typesetting of `date` (suppose `date` is just a year, but `eventdate` is an open-ended range, then the test would be true while `date` is typeset even though `date` is not an open-ended range).

Reply:
Good catch!